include/sys/types.h

```diff
@@ -251,10 +251,10 @@ typedef uint16_t     sa_family_t;
 
 #ifdef CONFIG_SYSTEM_TIME64
 typedef uint64_t     clock_t;
-typedef uint64_t     time_t;         /* Holds time in seconds */
+typedef int64_t      time_t;         /* Holds time in seconds */
 #else
 typedef uint32_t     clock_t;
-typedef uint32_t     time_t;         /* Holds time in seconds */
+typedef int32_t      time_t;         /* Holds time in seconds */
```

Comment:
this makes 32-bit time_t almost unusable. if we make this change, we should make a bold statement to users.